lib/hammer_cli_foreman/compute_resource/utils.rb

```diff
@@ -8,6 +8,21 @@ def self.get_image_uuid(compute_resource_id, image_id)
       )
     end
 
+    def self.get_hostgroup_compute_resource_id(hostgroup_id)
+      hostgroup = HammerCLIForeman.record_to_common_format(
+          HammerCLIForeman.foreman_resource(:hostgroups).call(:show, 'id' => hostgroup_id)
+      )
+      compute_resource_id = hostgroup['compute_resource_id']
+       if !hostgroup['compute_resource_name'].to_s.strip.empty? && compute_resource_id.nil?
+         compute_resource_id = HammerCLIForeman.record_to_common_format(
+            HammerCLIForeman.foreman_resource(:compute_resources).call(
+                :index, :search => "name = \"#{hostgroup['compute_resource_name']}\""
+            )
+         )['results'][0]['id']
```

Comment:
what if no matching resource are found?  `['results'][0]` will be `nil` i think?

Reply:
@tbrisker fixed :)